File: src/main/java/org/neonlabs/e2e/nethereum/Connection.cs
namespace NeonEndToEnd.org.neonlabs.e2e.nethereum
{
  using System;
  using Allure.Xunit;
  using dotenv.net.Utilities;
  using Nethereum.Web3;

  public class Connection
  {
    public static Web3 Connect()
    {
      return Steps.Step("Loading config", () => {
        Config.Init();

        // TODO: logging
        Console.WriteLine("===================");
        Console.WriteLine(Config.ProxyUrl);

        var web3 = new Web3(Config.ProxyUrl);
        // TODO: logging
        Console.WriteLine(web3);
        return web3;
      });
    }
  }
}

File: src/main/java/org/neonlabs/e2e/web3j/Connection.java
package org.neonlabs.e2e.web3j;

import io.qameta.allure.Step;
import org.web3j.protocol.Web3j;
import org.web3j.protocol.http.HttpService;

/**
 * Returns a connection to blockchain.
 */
public class Connection {
  /**
   * Creates a connection to blockchain.
   */
  @Step
  public Web3j createConnection() {
    var config = new Config();
    var web3 = Web3j.build(new HttpService(config.getProxyUrl()));
    return web3;
  }
}
